src/Vault.sol

```diff
@@ -460,6 +460,31 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver, ICDP {
         return this.onERC721Received.selector;
     }
 
+    function decreaseLiquidity(INonfungiblePositionManager.DecreaseLiquidityParams calldata params)
+        external
+        returns (uint256 amount0, uint256 amount1)
+    {
+        _requireVaultOwner(vaultIdByNft[params.tokenId]);
+
+        (amount0, amount1) = positionManager.decreaseLiquidity(params);
```

Comment:
Why there is no check, if position is healthy?

Reply:
This call can change capital estimation based on the chainlink oracle. As I can see capital estimation can only increase, but a check on the vault health would ensure it

Reply:
I think we added a test case which confirms that the price estimation remains exactly the same.
https://github.com/zkBob/stable-nft/blob/484234f0c8157f0eba25c5b9b83484c495b455cf/test/Vault.t.sol#L713-L729